### PR TITLE
render amenity=police area

### PR DESCRIPTION
**Pull request by RedAuburn**
basic fix for https://github.com/organicmaps/organicmaps/issues/3445

clear / vehicle
image1

### CODE REVIEW

**Comment by pastk** [comment]
Why this choice of color? Looks similar to airports and industrial areas...

Let's discuss what should it mean. Some kind of restriction? Alert?
Or just highlight amenity's area?

Note that we've added prison areas recently (#6325) in grey-ish. And we use red pattern for military zones.

File: data/styles/vehicle/include/priorities_4_overlays.prio.txt [hunk 1]
```diff
@@ -701,7 +701,7 @@ amenity-fast_food                                   # icon z17- (also has captio
 
 amenity-bank                                        # icon z17- (also has caption(optional) z17-)
 amenity-parcel_locker                               # icon z17- (also has caption(optional) z17-)
-amenity-police                                      # icon z17-
+amenity-police                                      # icon z17- (also has area z12-)
```
Comment (pastk) [inline data/styles/vehicle/include/priorities_4_overlays.prio.txt:704]: I think the area fill is not needed / not useful for a driver.
E.g. we don't use area fills for unis, hospitals...

File: data/styles/clear/include/Basemap.mapcss [hunk 2]
```diff
@@ -459,10 +459,14 @@ area|z15-16[amenity=parking][parking=lane],
 area|z15-16[amenity=parking][parking=street_side],
 {fill-color: none;}
 
+/* Other */
+area|z12-[amenity=police]
```
Comment (pastk) [inline data/styles/clear/include/Basemap.mapcss:463]: Let's make it z17- just like the icon.

Re the #2856 idea
1. we haven't decided on it for sure (need more experimentation)
2. when/if we decide it'd be better to migrate all areas at once

**Comment by biodranik** [comment]
What is the value for map users to see the police area? Does anyone (except OSMers) really need it?

**Comment by pastk** [comment]
> What is the value for map users to see the police area? Does anyone (except OSMers) really needs it?

- easier to spot when one needs it (to visit a station or to choose a place to stay closer to the station...)
- often such grounds are restricted access, hence to warn (similar to military zones and prisons)

**Comment by pastk** [comment]
> Why this choice of color? Looks similar to airports and industrial areas...
> 
> Let's discuss what should it mean. Some kind of restriction? Alert? Or just highlight amenity's area?
> 
> Note that we've added prison areas recently (#6325) in grey-ish. And we use red pattern for military zones.

@organicmaps/styles any ideas of a different fill color?

**Comment by euf** [comment]
In this case I'm not sure about purple color. It's already used for airports, and there is no cognitive links with it or aesthetic reasons to use it again. Also this purple is too eye-catching. While airports are important in terms of city topography and it's okay for them to be prominent, police stations are not that much of a landmarks for most of use cases.

I'd recommend trying the same hatching as for prisons #6325 because
- both prisons and police areas are related to law enforcement 
- hatching already represents somewhat limited access to these areas 
- it's not "dangerous", like military, so using gray instead of read makes more sense
- even if you believe police areas should be easily noticeable, they shouldn't have more visual accent than military areas

**Review by euf — CHANGES_REQUESTED** [review]
In this case I'm not sure about purple color. It's already used for airports, and there is no cognitive links with it or aesthetic reasons to use it again. Also this purple is too eye-catching. While airports are important in terms of city topography and it's okay for them to be prominent, police stations are not that much of a landmarks for most of use cases.

I'd recommend trying the same hatching as for prisons #6325 because
- both prisons and police areas are related to law enforcement 
- hatching already represents somewhat limited access to these areas 
- it's not "dangerous", like military, so using gray instead of read makes more sense
- even if you police areas should be noticeable, they shouldn't have more visual accent than military 

If gray hatching looks suboptimal, grayish blue tint `#5d729c` could work. 

I got it by starting from Navy Blue `#000080` and adjusting Lightness to `55.21` to match prison `#000080` and adjusting Chroma to `0.07` to be about 1/3 of military `#e53935` in OKLCH color picker:
https://oklch.com/#55.21,0.07,264.05202063805507,100

**Comment by pastk** [comment]
Many amenities/POIs have areas, not just police stations. And many of them are not rendered by OM. E.g. zoos, resorts, marketplaces, events venues, etc.

The idea is to add a generic "amenity" fill, the actual amenity it belongs to should be clear by map context (e.g. we can tell apart kindergartens and unis, right? :) even though they use the same fill color)
I think we can use `@pedestrian_area`, e.g. its being used for `amenity=place_of_worship` already.

WDYT?

**Comment by pastk** [comment]
IMO visualized borders could be possibly confused with barriers (like fences) whereas it could be just a territory without borders..

**Pull request closed without merging**